targets/TARGET_STM/rtc_api.c

```diff
@@ -62,7 +62,18 @@ void rtc_init(void)
     while (LL_HSEM_1StepLock(HSEM, CFG_HW_RCC_SEMID)) {
     }
 #endif /* DUAL_CORE */
-#if MBED_CONF_TARGET_LSE_AVAILABLE
+#if RTC_FROM_HSE
+#define RTC_HSE_DIV (HSE_VALUE / RTC_CLOCK)
+#if RTC_HSE_DIV > 31
+#error "HSE value too high for RTC"
+#endif
+    (void)RCC_OscInitStruct;
+    PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RTC;
+    PeriphClkInitStruct.RTCClockSelection = (RCC_RTCCLKSOURCE_HSE_DIVX | RTC_HSE_DIV << 16);
```

Comment:
RCC_RTCCLKSOURCE_HSE_DIVX is not a STM32L4 macro for ex

Reply:
Correct, this seems to get complicated to support the whole family. Do you think it is worth investing time or should I keep my changes as a fork?

Reply:
We could accept some specific code for TARGET_STM32F4 for ex

Reply:
I added a check for the target.